plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java

```diff
@@ -0,0 +1,114 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi;
+
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorSplitSource;
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.HdfsEnvironment.HdfsContext;
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.connector.TableNotFoundException;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.plugin.hudi.HudiErrorCode.HUDI_CANNOT_OPEN_SPLIT;
+import static io.trino.spi.connector.SchemaTableName.schemaTableName;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+
+public class HudiSplitManager
+        implements ConnectorSplitManager
+{
+    private final HudiTransactionManager transactionManager;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final ExecutorService executor;
+    private final int maxSplitsPerSecond;
+    private final int maxOutstandingSplits;
+
+    @Inject
+    public HudiSplitManager(
+            HudiTransactionManager transactionManager,
+            HdfsEnvironment hdfsEnvironment,
+            @ForHudiSplitManager ExecutorService executor,
+            HudiConfig hudiConfig)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.executor = requireNonNull(executor, "executor is null");
+        this.maxSplitsPerSecond = requireNonNull(hudiConfig, "hudiConfig is null").getMaxSplitsPerSecond();
+        this.maxOutstandingSplits = hudiConfig.getMaxOutstandingSplits();
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        this.executor.shutdown();
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle tableHandle,
+            SplitSchedulingStrategy splitSchedulingStrategy,
```

Comment:
`SplitSchedulingStrategy splitSchedulingStrategy,` can be removed because it has been (recently) deprecated.

Reply:
noted